process_report/tests/unit_tests.py

```diff
@@ -175,3 +176,65 @@ def test_export_pi(self):
         self.assertNotIn('ProjectA', pi_df['Project - Allocation'].tolist())
         self.assertNotIn('ProjectB', pi_df['Project - Allocation'].tolist())
         self.assertNotIn('ProjectC', pi_df['Project - Allocation'].tolist())
+
+
+class TestGetInstitute(TestCase):
+
+    def setUp(self):
+
+        data = {
+            'Manager (PI)': ['quanmp@bu.edu', 'c@mclean.harvard.edu', 'b@harvard.edu', 'fake', 'pi@northeastern.edu'],
+            'Project - Allocation': ['ProjectA', 'ProjectB', 'ProjectC', 'ProjectD', 'ProjectE'],
+            'Answer': ["Boston University", "McLean Hospital", "Harvard University", "", "Northeastern University"]
+        }
+        self.data = pandas.DataFrame(data)
+
+    def test_get_pi_institution(self):
+        for i, row in self.data.iterrows():
+            institution_name = process_report.get_institution_from_pi(row['Manager (PI)'])    
+            self.assertEqual(institution_name, row['Answer'])
+
+@skipIf(process_report.apply_credits_0002 not in process_report.applied_credits, "Skipping test for credit 0002 because credit not enabled")
+class TestCredit0002(TestCase):
+    def setUp(self):
+
+        data = {
+            'Manager (PI)': ['PI1', 'PI1', 'PI2', 'PI3', 'PI4', 'PI4'],
+            'Project - Allocation': ['ProjectA', 'ProjectB', 'ProjectC', 'ProjectD', 'ProjectE', 'ProjectF'],
+            'Cost': [10, 100, 10000, 5000, 800, 1000]
+        }
+        self.dataframe = pandas.DataFrame(data)
+        old_pi = ['PI2', 'PI3']
```

Comment:
I am probably missing something. do we fill the old_pi file with previous values so we can test that it subtracts from this value? I also want to make sure that it only is providing this discount for 2 months no more. In other words, as soon as a PI has been on the list for 2 months even if they have not gone through the $1000 they don't get any more.

Reply:
@joachimweyl The way I have implemented for the "New PI" credit is applied is that:
You provide a file containing a list of old PIs (PIs who will not receive the "New PI" credit in this invoice). You must create and maintain this "Old PI file" yourself. My code so far doesn't contain any logic about how long the PI has joined NERC or etc. It currently does not get information from any external data sources, other than the files the person running the script has provided.

Reply:
to automate the credit code 0002 we need to:
1) have an existing PI list (new_pi.csv) that has 2 columns 1 is the pi address and 2 is age (filled with 1, 2, old) or something similar
   1) this could be combined with a new column that tracks how much of the 1000 they have used up
2) update the  list based on the current data file
3) provide credit based on the second column

Reply:
@joachimweyl The creation of a `new_pi.csv` would be a seperate problem. With your permission, I could download all the invoice data we have since as far back as you'd like, and write a script to generate the `new_pi.csv`. After that I can include logic to update `new_pi.csv`.

I assume this probably mean `new_pi.csv` has to be version controlled and put in this repo? And we'd have to submit a monthly PR to update the one file. I not exactly certain if this is the best way to do thing, and would rather have opinions from @knikolla and @naved001.

Reply:
agreed, I would rather this automatically update when the script is run, not requiring a PR to update if possible.